locator/tablets.cc

```diff
@@ -867,8 +954,9 @@ std::unordered_set<sstring> tablet_aware_replication_strategy::recognized_tablet
 
 future<effective_replication_map_ptr> tablet_aware_replication_strategy::do_make_replication_map(
         table_id table, replication_strategy_ptr rs, token_metadata_ptr tm, size_t replication_factor) const {
-    return make_ready_future<effective_replication_map_ptr>(
-            seastar::make_shared<tablet_effective_replication_map>(table, std::move(rs), std::move(tm), replication_factor));
+    auto erm = seastar::make_shared<tablet_effective_replication_map>(table, std::move(rs), std::move(tm), replication_factor);
+    co_await erm->build_dc_replication_factor_map();
```

Comment:
Can we detect that there is no tablets in transition and do not populate anything? If the array is not populated we return values according to RF.

Reply:
IIRC this won't work because when applying `mixed_change` or snapshot there is a short period of time where the schema is updated, but topology is not updated yet. So we construct e_r_m based on updated schema (e.g. RF reduced from 1 to 0) but old topology (tablets are still not in transition state). This state in practice holds for less than a second -- it is basically happening between these two co_awaits:
```cpp
    [&] (mixed_change& chng) -> future<> {
        co_await _mm.merge_schema_from(netw::messaging_service::msg_addr(std::move(cmd.creator_addr)), std::move(chng.mutations));
        co_await _ss.topology_transition();
```

but it's still happening and if we allowed it, the bug is still present.

Reply:
If the goal is to reduce lookups and memory in the case of shared RF, we could compute the shared RF map from erm if we notice that all tablets have the same replica count.

Reply:
Either way we should probably fix `mixed_change` so it reloads topology and schema in isolated fashion. (I believe @nuivall's work will address it)